src/diff.c

```diff
@@ -579,6 +591,12 @@ diff_alloc_new(tabpage_T *tp, diff_T *dprev, diff_T *dp)
 	tp->tp_first_diff = dnew;
     else
 	dprev->df_next = dnew;
+
+    CLEAR_FIELD(dnew->df_lnum);
+    CLEAR_FIELD(dnew->df_count);
```

Comment:
perhaps use `ALLOC_CLEAR_ONE(diff_T)` above and then get rid for `CLEAR_FIELD` here?

Reply:
Done.